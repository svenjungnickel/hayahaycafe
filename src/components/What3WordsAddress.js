import React from 'react';
import PropTypes from 'prop-types';
import { graphql, StaticQuery } from 'gatsby';
import Image from './Image';
import What3WordsAddressStyles from '../styles/components/What3WordsAddress.module.scss';

const What3Words = ({ what3WordsAddress, what3WordsIcon }) => (
    <a
        href={`https://what3words.com/${what3WordsAddress}`}
        target="_blank"
        rel="noreferrer"
        className={What3WordsAddressStyles.what3words}
    >
        <Image src={what3WordsIcon} alt="What3words address" />
        <p className={What3WordsAddressStyles.what3words__address}>///{what3WordsAddress}</p>
    </a>
);

What3Words.propTypes = {
    what3WordsAddress: PropTypes.string.isRequired,
    what3WordsIcon: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired,
};

const What3WordsAddress = () => (
    <StaticQuery
        query={graphql`
            query {
                globalSettings: settingsYaml {
                    location {
                        what3WordsAddress
                    }
                }
                what3WordsIcon: file(relativePath: { eq: "what3words-icon.png" }) {
                    childImageSharp {
                        fixed(width: 80, height: 80) {
                            ...GatsbyImageSharpFixed
                        }
                    }
                    publicURL
                }
            }
        `}
        render={({ globalSettings, what3WordsIcon }) => (
            <>
                {!!globalSettings && !!globalSettings.location && !!globalSettings.location.what3WordsAddress && (
                    <>
                        <h2>Map</h2>
                        <What3Words
                            what3WordsAddress={globalSettings.location.what3WordsAddress}
                            what3WordsIcon={what3WordsIcon}
                        />
                    </>
                )}
            </>
        )}
    />
);

export default What3WordsAddress;
