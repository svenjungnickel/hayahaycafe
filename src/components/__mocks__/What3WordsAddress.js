import React from 'react';

const What3WordsAddress = () => (
    <a href="what3WordsAddress">
        <img src="src" alt="what3WordsIcon" />
    </a>
);

export default What3WordsAddress;
