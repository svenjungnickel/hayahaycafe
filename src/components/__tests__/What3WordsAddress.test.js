import React from 'react';
import renderer from 'react-test-renderer';
import { StaticQuery } from 'gatsby';
import What3WordsAddress, { What3Words } from '../What3WordsAddress';

jest.mock('../Image');

describe('What3WordsAddress', () => {
    it('renders without required what3WordsAddress throws prop type error', () => {
        const renderWhat3Words = () => {
            const component = <What3Words />;
            renderer.create(component);
        };

        expect(renderWhat3Words).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid what3WordsAddress throws prop type error', () => {
        const renderWhat3Words = () => {
            const what3WordsAddress = 123;

            const component = <What3Words what3WordsAddress={what3WordsAddress} />;
            renderer.create(component);
        };

        expect(renderWhat3Words).toThrowError('Warning: Failed prop type');
    });

    it('does not render empty data', () => {
        const data = {};
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <What3WordsAddress />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('does not render empty globalSettings data', () => {
        const data = {
            globalSettings: {},
        };
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <What3WordsAddress />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('does not render empty location data', () => {
        const data = {
            globalSettings: {
                location: {},
            },
        };
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <What3WordsAddress />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders correctly', () => {
        const data = {
            globalSettings: {
                location: {
                    what3WordsAddress: 'what3WordsAddress',
                },
            },
        };
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <What3WordsAddress />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
