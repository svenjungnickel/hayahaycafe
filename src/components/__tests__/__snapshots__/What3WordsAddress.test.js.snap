// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`What3WordsAddress does not render empty data 1`] = `null`;

exports[`What3WordsAddress does not render empty globalSettings data 1`] = `null`;

exports[`What3WordsAddress does not render empty location data 1`] = `null`;

exports[`What3WordsAddress renders correctly 1`] = `
<div
  className="what3words"
>
  <h2>
    Map
  </h2>
  <a
    className="what3words__link"
    href="https://what3words.com/what3WordsAddress"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="What3words address"
      height={80}
      src="../../assets/what3words-icon.png"
      width={80}
    />
    <p
      className="what3words__address"
    >
      ///what3WordsAddress
    </p>
  </a>
</div>
`;
